gnssanalysis/gn_download.py

```diff
@@ -268,8 +292,13 @@ def check_n_download_url(url, dwndir, filename=False):
         out_f = _Path(dwndir) / filename
         download_url(url, out_f)
 
+    if uncomp:
+        _subprocess.run(["uncompress", f"{out_f}"])
```

Comment:
This functionality is a little unintuitive in its relationship to the `filename` argument to this function. If one were to specify the desired uncompressed filename as an argument the function would instead download the compressed to a file without a `.Z` extension, at which point I believe `uncompress` will not process it and the user will wind up with a compressed file with the name of the desired uncompressed file, which is almost certainly not what they expected.
This also means that this functionality has relatively unintuitive interactions with the "check" part of `check_n_download_url`. Because the check looks for the original filename in the case of a compressed file it's effectively looking for the name of the compressed file, so if you already have the uncompressed file you want the function will still download the uncompressed file, attempt to run `uncompress`, which will fail because `uncompress` won't overwrite the existing file (without user input) and so I think you will wind up with the (potentially old) original uncompresed file and the new compressed version both sitting on disk. This scenario would be most noticeable if you ran the function multiple times, where I believe the first time it would work as expected, the second time I think you would download the compressed file again and wind up with that living on disk and then subsequent runs might do nothing but attempt to run `uncompress` on the previously downloaded compressed file and fail. I think the scenario where you have the compressed file on disk already but not the uncompressed version actually works as expected, you don't download but you do `uncompress`.